src/main/java/com/redhat/devtools/lsp4ij/client/LanguageClientImpl.java

```diff
@@ -218,12 +218,16 @@ public CompletableFuture<List<Object>> configuration(ConfigurationParams params)
      * must return an instance of {@link JsonObject} but this method can be overridden to return another structure.
      * </p>
      *
-     * @param section the section.
-     * @return the settings retrieved by the given section and null otherwise.
+     * @param section the section. In the base implementation, a null section indicates a request for the entire
+     *                settings object.
+     * @return the settings retrieved by a valid section and null otherwise.
      */
     protected Object findSettings(String section) {
```

Comment:
I did not do `@Nullable String section` in the parameters here, because that indicates subclasses should also accept `null` sections.

Reply:
If method can receive null I think it should be better to use @nullable annotation, no?

Reply:
`findSettings` is `protected`, so if we add `@Nullable` here, I understand this to mean all third-party plugins which have subclassed `LanguageClientImpl` and overridden `findSettings` to also be `@Nullable`.

I don't know the downstream (subclassing) behaviour of adding `@Nullable` in Java here - in other languages, this would violate liskov substitution and cause compile-time errors. If you're not concerned about this, I can add `@Nullable`.

Reply:
I have never seen some project which extends findSettings, so I would prefer providing a clean API. Si please add @Nullable